charts/flyte-core/templates/admin/deployment.yaml

```diff
@@ -72,6 +72,10 @@ spec:
             name: clusters-config-volume
           - mountPath: /etc/secrets/
             name: admin-secrets
+          {{- if gt (len .Values.configmap.clusters.labelClusterMap) 0 }}
+          - mountPath: /var/run/credentials
+            name: cluster-credentials
```

Comment:
but this is a volumeMount right?  doesn't that mean there needs to be a volume declared with the same name in the Pod?

Reply:
that will mount the secret to that path, no?

Reply:
Also no, since this is an init container and the main container is already declaring the volume for the secret

Reply:
oh i see it's in additionalVolumes

Reply:
but wait shouldn't that come through in the additional mounts?
```
  additionalVolumeMounts:
  - name: cluster-credentials
    mountPath: /var/run/credentials
```
which gets injected here: https://github.com/flyteorg/flyte/blob/a071bade39bdad80fff042c235a1c3c046a82a09/charts/flyte-core/templates/admin/deployment.yaml#L137C37-L137C59

Reply:
@gdabisias I just found that if we change the process to, instead of creating a new secret,  editing the existing `flyte-admin-secrets` and adding the data plane cluster token and cert there, the `syncresources` Pod works just fine. 
Right, no surprises there.

Reply:
@davidmirror-ops as you mentioned in the other comment, we should have a separate secret

Reply:
@wild-endeavor no because here we are updating the syncresouces init container and not the admin one. For the admin one, we add the secret mounting point to the values-eks.yaml

Reply:
should we reference `Values.flyteadmin.additionalVolumeMounts` also instead of directly adding the mountpath?

The issue is that this only works if the user specifies that string in the values file for the additional volumes. if they use a different string than "cluster-credentials" then this won't work right?

Reply:
yes, it won't work. My idea was to fix things according to the guide, but we can also just add that to the values-eks.yaml.
Up to you, I don't mind. Either way, we should remove this secret creation and mounting completely and it should be part of the general chart, with the user only adding the secret value itself

Reply:
@wild-endeavor I checked again and we can't sure that field, since here we are mounting things for the sync-cluster-resources init container, which is not the admin container.
the admin container is already loading the volume in the additional section that users have to add to the deployment to make things work, e.g.
```
flyteadmin: {
			additionalVolumes: [{
				name: "cluster-credentials"
				secret: secretName: "cluster-credentials"
			}]
			additionalVolumeMounts: [{
				name:      "cluster-credentials"
				mountPath: "/var/run/credentials"
			}]
```

but this can't include the values for the init container (if you mount something for the flyteadmin container, it won't be exported automatically to the sync-cluster-resources container

Reply:
I made the change but notice that we are mounting anything specified in the additional volumes section, so there might be also some other stuff (Still better than what we had before and I don't see why something mounted by the admin container should not be mounted by it's init container)